src/Bicep.Core/Registry/OciArtifactRegistry.cs

```diff
@@ -54,23 +54,23 @@ public sealed class OciArtifactRegistry : ExternalArtifactRegistry<OciModuleRefe
 
         public string CacheRootDirectory => this.features.CacheRootDirectory;
 
-        public override RegistryCapabilities GetCapabilities(OciModuleReference reference)
+        public override RegistryCapabilities GetCapabilities(OciArtifactReference reference)
         {
             // cannot publish without tag
             return reference.Tag is null ? RegistryCapabilities.Default : RegistryCapabilities.Publish;
         }
 
-        public override ResultWithDiagnostic<ArtifactReference> TryParseArtifactReference(string? aliasName, string reference)
+        public override ResultWithDiagnostic<ArtifactReference> TryParseArtifactReference(string? aliasName, ArtifactType artifactType, string reference)
```

Comment:
[nit] This is perhaps a personal prefernce and difficult to define as a rule per-se. In general, I try to put the things which are "more static" or have specific types earlier in the argument list - e.g. rather than:
```csharp
public void DoSomething(string value, ISomethingDoer doer)
```
I would usually go for:
```csharp
public void DoSomething(ISomethingDoer doer, string value)
```

Here because the parsing is going to depend on the artifact type, and the artifact type is generally fixed w.r.t the function invocation, I'd prefer;
```csharp
public override ResultWithDiagnostic<ArtifactReference> TryParseArtifactReference(ArtifactType artifactType, string? aliasName, string reference)
```

FWIW, VS has good refactoring tools to do this (see [here](https://learn.microsoft.com/en-us/visualstudio/ide/reference/change-method-signature?view=vs-2022#how-to))